gammapy/detect/lima.py

```diff
@@ -2,13 +2,68 @@
 import copy
 import logging
 import numpy as np
+from astropy.coordinates import Angle
+from astropy.convolution import Tophat2DKernel
 from gammapy.stats import significance, significance_on_off
+from gammapy.cube import MapDataset, MapDatasetOnOff
 
-__all__ = ["compute_lima_image", "compute_lima_on_off_image"]
+__all__ = [
+    "SignificanceMapEstimator",
+    "compute_lima_image",
+    "compute_lima_on_off_image",
+]
 
 log = logging.getLogger(__name__)
 
 
+class SignificanceMapEstimator:
+    """Computes correlated excess, significance for MapDatasets
+
+
+    Parameters
+    ----------
+    correlation_radius : ~astropy.coordinate.Angle
+        correlation radius to use
+    """
+
+    def __init__(self, correlation_radius):
+        self.radius = Angle(correlation_radius)
+
+    def run(self, dataset):
+        """Compute correlated excess, Li & Ma significance and flux maps
+
+        This requires datasets with only 1 energy bins (image-like).
+        Usually you can obtain one with `dataset.to_image()`
+
+
+        Parameters
+        ----------
+        dataset : `~gammapy.cube.MapDataset` or `~gammapy.cube.MapDataset`
+            input dataset
+        """
+        if not isinstance(dataset, MapDataset) and not isinstance(
```

Comment:
`isinstance()` can be used with a tuple like so: `isinstance(dataset, (MapDataset, MapDatasetOnOff))`. It also takes inheritance into account, this means `isinstance(dataset_on_off, MapDataset)` will also return `True`. I would propose to simplify here a bit.

Reply:
Right. Done.